File: src/services/api/orchestrationAdmin.ts
import { apiRequest } from './client';

/** Communication-cap policy — caps how often a single contact may be reached
 *  per (tenant, app) within a rolling window. Shape mirrors the Phase 2
 *  `CommCapPolicyRead` backend schema exactly. */
export interface CommCapPolicy {
  id: string;
  tenantId: string;
  appId: string;
  maxCount: number;
  windowSeconds: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  updatedByUserId: string | null;
}

export interface CommCapPolicyWrite {
  tenantId: string;
  appId: string;
  maxCount: number;
  windowSeconds: number;
  isActive: boolean;
}

export async function listCommCapPolicies(): Promise<CommCapPolicy[]> {
  return apiRequest<CommCapPolicy[]>('/api/admin/orchestration/comm-cap/list');
}

export async function upsertCommCapPolicy(
  body: CommCapPolicyWrite,
): Promise<CommCapPolicy> {
  return apiRequest<CommCapPolicy>('/api/admin/orchestration/comm-cap', {
    method: 'PUT',
    body: JSON.stringify(body),
  });
}

File: backend/app/schemas/orchestration_admin.py
"""API shapes for the orchestration administration surface (Phase 2+).

Currently scoped to communication-cap policy management. Future Phase 3/4
admin surfaces (e.g. retry policy, escalation policy) extend this module.
"""
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import Field

from app.schemas.base import CamelModel, CamelORMModel


class CommCapPolicyRead(CamelORMModel):
    id: UUID
    tenant_id: UUID
    app_id: str
    max_count: int
    window_seconds: int
    is_active: bool
    created_at: datetime
    updated_at: datetime
    updated_by_user_id: UUID | None = None


class CommCapPolicyWrite(CamelModel):
    tenant_id: UUID
    app_id: str
    max_count: int = Field(gt=0)
    window_seconds: int = Field(gt=0)
    is_active: bool = True
